geometry/r3_element.hpp

```diff
@@ -23,41 +25,74 @@ using quantities::Quotient;
 template<typename Scalar>
 struct SphericalCoordinates;
 
-// An |R3Element<Scalar>| is an element of Scalar³. |Scalar| should be a vector
-// space over ℝ, represented by |double|. |R3Element| is the underlying data
-// type for more advanced strongly typed structures suchas |Multivector|.
-template<typename Scalar>
+template<typename ScalarX,
+         typename ScalarY = ScalarX,
+         typename ScalarZ = ScalarX>
+struct R3Element;
+
+template<typename T>
+struct enable_if_normed : not_constructible {};
+
+template<typename T>
+struct enable_if_normed<R3Element<T, T, T>> : not_constructible {
+  using Scalar = T;
+};
+
+// An |R3Element| is an element of ℝ³. |ScalarX|, |ScalarY| and |ScalarZ| must
+// be 1-dimensional vector spaces over ℝ, typically represented by |Quantity|
+// or |double|.  If the three scalars are homogeneous (in the dimensional
+// analysis sense, not in the projective sense), then the resulting vector space
+// is also normed.
+// |R3Element| is the underlying data type for more advanced strongly typed
+// structures such as |Multivector|.
+template<typename ScalarX, typename ScalarY, typename ScalarZ>
```

Comment:
As discussed, let's not go there, we're not using the vector space structure in the projective plane.